app/src/main/java/tech/ula/ServerService.kt

```diff
@@ -80,17 +85,100 @@ class ServerService : Service() {
             }
         }
 
-        return Service.START_STICKY
+        val action = intent?.action
+        when (action) {
+            "autostart" -> {
+                autoStartSessions()
+            }
+            "exit" -> {
+                stopServiceAndSessions()
+            }
+            "wakelock" -> {
+                acquireWakeLocks()
+            }
+            "wakerelease" -> {
+                releaseWakeLocks()
+            }
+        }
+        return START_STICKY
+    }
+
+    /**
+     * Fetch all sessions marked to start on boot and start them if they are not already active.
+     */
+    private fun autoStartSessions() {
+        CoroutineScope(Dispatchers.Default).launch {
```

Comment:
I will look into this and change the implementation accordingly.

Reply:
@MatthewTighe I have been thinking about the different modes of setup that a user might need and came up with this set of options.
I will follow the same layout flow of the filesystem_edit fragment and make this options available under the "Show advanced options" toggle.
Some validation will be required for some cases (eg: There is no need to launch the client if the server is disabled.)

![image](https://user-images.githubusercontent.com/4238591/57003338-c1725f80-6bbd-11e9-8967-da0017fd8130.png)

Reply:
One more thing to think about here: There's two places that a command being executed on "start" could be executed if a server is also being started. 
1. _Before_ starting the server. I think the best way to run the command in this case would be to call it _from_ the script that starts the server. This would be a little work on our end, but it would mean that the name of the command (or better the location of a script containing any number of commands) would need to be passed as an environment variable to the `BusyboxExecutor`.
2. _After_ starting the server and having a client connect. This one is easier, and can just mirror the way that we run scripts when starting apps. That is to put them as the last entry in `/etc/profile.d` such that they execute automatically on client login.

Reply:
Will look into this after improving the current implementation.